platform/linux-generic/odp_traffic_mngr.c

```diff
@@ -2589,19 +2593,33 @@ int odp_tm_capabilities(odp_tm_capabilities_t capabilities[] ODP_UNUSED,
 		per_level_cap->tm_node_dual_slope_supported = true;
 		per_level_cap->fair_queuing_supported       = true;
 		per_level_cap->weights_supported            = true;
+		per_level_cap->tm_node_threshold            = true;
 	}
 
 	return 1;
 }
 
+int odp_tm_egress_capabilities(odp_tm_capabilities_t *capabilities,
+			       const odp_tm_egress_t *egress)
+{
+	int ret;
+
+	(void)egress;
+
+	ret = odp_tm_capabilities(capabilities, 1);
```

Comment:
Now that the API says that pktio type egresses without TM enabled cause capability to indicate no TM support, the implementation should be changed accordingly. So you need to check if get_pktio_entry()->s.param.out_mode is ODP_PKTOUT_MODE_TM before calling the common capability function.